transports/email/src/test/java/org/mule/transport/email/functional/SmtpTransportConfigTestCase.java

```diff
@@ -0,0 +1,76 @@
+package org.mule.transport.email.functional;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+import org.mule.api.MuleException;
+import org.mule.api.endpoint.OutboundEndpoint;
+import org.mule.api.transport.MessageDispatcher;
+import org.mule.endpoint.MuleEndpointURI;
+import org.mule.transport.email.SmtpMessageDispatcher;
+import org.mule.transport.email.SmtpMessageDispatcherFactory;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.CountDownLatch;
+
+public class SmtpTransportConfigTestCase extends AbstractEmailFunctionalTestCase
+{
+
+    private static String user;
+
+    private static final String TEST_USER = "test%40@test.com";
+
+    private static final CountDownLatch latch = new CountDownLatch(1);
+
+    public SmtpTransportConfigTestCase(ConfigVariant variant, String configResources)
+    {
+        super(variant, STRING_MESSAGE, "smtp", configResources);
+    }
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> parameters()
+    {
+        return Arrays.asList(new Object[][]{
+                {ConfigVariant.FLOW, "smtp-transport-config-test.xml"}
+        });
+    }
+
+    @Test
+    public void testSend() throws Exception {
+        runFlow("endpointTransportConfigFlow");
+        latch.await();
+        assertThat(user, is(TEST_USER));
```

Comment:
Shouldn't the password be verified as well? It would be better to add special characters in the password and add an assert involving that also. Can it be retrieved from the transport's uri in the same way?

Reply:
It is not verified as the password is not visible from the SmtpTransport.